ui/components/multichain/edit-accounts-modal/edit-accounts-modal.tsx

```diff
@@ -17,65 +22,180 @@ import {
   ModalHeader,
   Checkbox,
   Box,
+  ModalFooter,
+  ButtonPrimary,
+  ButtonPrimarySize,
 } from '../../component-library';
 import { AccountListItem } from '..';
 import { MergedInternalAccount } from '../../../selectors/selectors.types';
 import { mergeAccounts } from '../account-list-menu/account-list-menu';
+import {
+  addMorePermittedAccounts,
+  removePermissionsFor,
+  removePermittedAccount,
+} from '../../../store/actions';
+import { SubjectsType } from '../pages/connections/components/connections.types';
+
+const defaultAllowedAccountTypes = [EthAccountType.Eoa, EthAccountType.Erc4337];
 
 type EditAccountsModalProps = {
   onClose: () => void;
+  onClick: () => void;
   allowedAccountTypes?: KeyringAccountType[];
 };
 
-const defaultAllowedAccountTypes: KeyringAccountType[] = [
-  EthAccountType.Eoa,
-  EthAccountType.Erc4337,
-];
-
-export const EditAccountsModal = ({
+export const EditAccountsModal: React.FC<EditAccountsModalProps> = ({
   onClose,
+  onClick,
   allowedAccountTypes = defaultAllowedAccountTypes,
-}: EditAccountsModalProps) => {
+}) => {
   const t = useI18nContext();
   const accounts = useSelector(getUpdatedAndSortedAccounts);
   const internalAccounts = useSelector(getInternalAccounts);
+  const dispatch = useDispatch();
 
   const mergedAccounts: MergedInternalAccount[] = useMemo(() => {
     return mergeAccounts(accounts, internalAccounts).filter(
       (account: InternalAccount) => allowedAccountTypes.includes(account.type),
     );
-  }, [accounts, internalAccounts, allowedAccountTypes]); // Add allowedAccountTypes to dependency array
+  }, [accounts, internalAccounts, allowedAccountTypes]);
+
+  const activeTabOrigin = useSelector(getOriginOfCurrentTab);
+  const subjects = useSelector(getPermissionSubjects);
+  const connectedAccounts = useSelector((state: any) =>
```

Comment:
If possible let's avoid the use of `any`

Reply:
Replaced it with   const connectedAccounts = useSelector((state) =>
    getOrderedConnectedAccountsForConnectedDapp(state, activeTabOrigin).filter(
      (account: InternalAccount) => isEvmAccountType(account.type),
    ),
  );